### PR TITLE
fixed errors found in cursor implementation

### PR DESCRIPTION
**Pull request by soraia653**
## Description

While trying to implement this cursor I found many issues with the code provided in the tutorial.

- Removed usage of attributes.
- Removed usage of map (replaced with list comprehension).
- Added missing field ID to User class.
- Added method to User class.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* #2553

## Checklist

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/guides/pagination/cursor-based.md [hunk 1]
```diff
@@ -58,25 +58,40 @@ Let us model our schema like this:
...
         description="The occupation of the user."
     )
 
+
     age: int = strawberry.field(
         description="The age of the user."
     )
 
+    @staticmethod
+    def from_row(row: Dict[str, Any]):
```
Comment (patrick91) [inline docs/guides/pagination/cursor-based.md:87]: let's also type the return value 😊 

```suggestion
    def from_row(row: Dict[str, Any]) -> "User":
```

File: docs/guides/pagination/cursor-based.md [hunk 2]
```diff
@@ -145,18 +160,33 @@ user_data = [
...
```
Comment (patrick91) [inline docs/guides/pagination/cursor-based.md:182]:
```suggestion
    def from_row(row: Dict[str, Any]) -> "User":
```

File: docs/guides/pagination/cursor-based.md [hunk 3]
```diff
@@ -270,6 +304,15 @@ class User:
...
```
Comment (patrick91) [inline docs/guides/pagination/cursor-based.md:308]:
```suggestion
    def from_row(row: Dict[str, Any]) -> "User":
```

File: docs/guides/pagination/cursor-based.md [hunk 4]
```diff
@@ -377,6 +425,15 @@ class User:
...
```
Comment (patrick91) [inline docs/guides/pagination/cursor-based.md:429]:
```suggestion
    def from_row(row: Dict[str, Any]) -> "User":
```

File: docs/guides/pagination/cursor-based.md [hunk 5]
```diff
@@ -409,7 +466,7 @@ class Query:
           user_id = 0
 
         # filter the user data, going through the next set of results.
-        filtered_data = map(lambda user: user.id > user_id, user_data)
+        filtered_data = [user for user in user_data if user['id'] >= user_id]
```
Comment (patrick91) [inline docs/guides/pagination/cursor-based.md:469]: 👍

File: docs/guides/pagination/cursor-based.md [hunk 6]
```diff
@@ -418,14 +475,13 @@ class Query:
...
-          next_cursor = encode_user_cursor(id=last_user.id)
+          next_cursor = encode_user_cursor(id=last_user['id'])
         else:
           # We have reached the last page, and
           # don't have the next cursor.
           next_cursor = None
 
-        # type cast the sliced data.
-        sliced_users = cast(List[UserType], sliced_users)
+        sliced_users = [User.from_row(x) for x in sliced_users]
```
Comment (patrick91) [inline docs/guides/pagination/cursor-based.md:484]: 💯

**Review by patrick91 — APPROVED** [review]
This looks great to me! thank you so much 😊

I've left a minor change request, the rest looks great!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**